### ISSUE TITLE
Question in regard `prepareTestResults.m`

### ISSUE THREAD

**Issue by nitba** [anibali/eval-mpii-pose#10]
Hi @anibali ,

I saw their codes you referenced for plotting, sth confusing is that for validation and testing they used the same index from `MPII` and as it is mentioned in `stacked hourglass paper` they provide results on validation because test annotation is not available , and you have provided `annotation folder` which separate `train, val, test` index, I confused a little, would you please clarify me?

**Comment by anibali** [comment]
You'll find that the `test.h5` file doesn't actually contain meaningful joint annotations, but does contain some other information like person centre and scale and indexing details. So it can't be used to calculate accuracy, but is handy for preparing results for submission.

**Comment by nitba** [comment]
Hi @anibali 
Another question is that when I see annotation provided by  `pose -pytorch`, it has `train `attribute to separate `val` and `train`, 
and when I saw the `stacked hour glass` data in his github it has test and train and val h files which has `image name` to separate them, as you know in each image there may be multi  persons , so  all of the person specified by image name are used for testing or training?

**Comment by anibali** [comment]
The original dataset has a train and test set, but no validation set. The istrain attribute separates those two sets. A little while after the dataset was released, authors (Tompson et al, I believe) further split a validation set from the training set in order to run more experiments without submitting to the leaderboard. This particular validation set has since been reused by multiple other papers.

Now the three sets (train, valid, and test) do not overlap in terms of specific images used---all people in an image belong to the same set, as you suggested. However, as far as I am aware there can be overlap in terms of the videos from which the images originated from. So I think it is possible for a person's likeness to appear in multiple images, which in turn may have been split into different sets.

I hope that this answers your question.

**Comment by nitba** [comment]
Thank you so much @anibali 
Another point which is confusing for me in `MPI` is that I saw some papers talked about `Single Person` data and `Multi person Data` in `MPI`, and I just see one `annotation matrix ` for all `MPI` data which has various number of people, what does this separation  means, are they talking new branch of  `MPI` data, or they just separated the images of `single person` while talking about `MPI` single person?  if so , Do you think what is the reason for that , they always can get the bounding box of people to treat that as a single person

**Comment by anibali** [comment]
The multi-person task is slightly more difficult since you are not allowed to use some of the annotation data, like approximate person centre and scale. See http://human-pose.mpi-inf.mpg.de/#evaluation. Other than that, the data is the same (same images and same Matlab annotation files).

**Comment by nitba** [comment]
Thank you @anibali 
I used `Convolutional Pose Machine` , `genJSON` to produce `val`, `train ` splits,  in `MPII`it has different number of samples with respect to `pose pytorch` github,Would you pleas provide files that you produce the `train,test,val` `h5` splits to check,

**Comment by anibali** [comment]
As stated in the Readme, those h5 annotation files were sourced from https://github.com/princeton-vl/pose-hg-train/tree/master/data/mpii/annot. I did not create them myself.

**Comment by nitba** [comment]
Hi @anibali 
about evaluation, as you know `during training` we also train on the joints `which have annotation in image but are occluded or not visible because of other stuff  `  by assuming them as they are visible, right? Do I get it correctly according to `stacked hourglass` ?
During evaluation  Do we consider these joints as visible too?

**Comment by anibali** [comment]
If memory serves correctly, joints with coordinates that are zeros are excluded from evaluation, but other joints are included.

**Comment by nitba** [comment]
Hi @anibali ,
for evaluation it is said in `MPI` dataset , that it is allowed to use  `scale` and `objpose` for normalization, but in addition to that we used the for preprocessing, in all single person pose estimation is it allowed to use the information of `scale` and `person pose` or `bounding box`?

**Comment by anibali** [comment]
You can use the scale and centre to create a bounding box. You can't use the pose as far as I know.

**Comment by nitba** [comment]
you mean that for evaluation in testing, we can use `center, bbox, scale`  like training for single person pose estimation?

**Comment by anibali** [comment]
For single person pose estimation---yes. Otherwise they would simply not provide those annotations for the test data.
